src/gallery/gestures.js

```diff
@@ -0,0 +1,208 @@
+const scaleMin = 1
+const scaleMax = 2
+const clientWidth = window.innerWidth
+const clientHeight = window.innerHeight
+const temp = {
+	screenX: null,
+	originalMarginLeft: null,
+	currentMarginLeft: null,
+	originalTransition: null,
+	imgOriginalTransition: null,
+	durationStart: null,
+	translateX: 0,
+	translateY: 0,
+	clientX: null,
+	clientY: null
+}
+let evCache = []
+let prevDiff = -1
+let zoomedIn = false
+
+const grabImageFromEvent = ( e ) => {
+	return e.target.nodeName === 'IMG' ? e.target : e.target.querySelector( 'img' )
+}
+
+const grabScaleFromTransform = ( transform ) => {
+	return Number( transform.slice( transform.indexOf( 'scale' ) + 6, -1 ) )
+}
+
+const isInvalidEvent = ( e, items, current, captionText, prefixClassname ) => {
+	return e.target.className === captionText &&
+			items[ current ].querySelector( `.${prefixClassname}-item-caption-expand-cue` ) ||
+			e.target.className === captionText ||
+			e.pointerType !== 'touch' ||
+			e.target.className.indexOf( 'slider-button' ) > -1
+}
+
+const isImgZoomedIn = () => {
+	return zoomedIn
+}
+
+const getFingerAmount = () => {
+	return evCache.length
+}
+
+const removeEvent = ( e ) => {
+	for ( let i = 0; i < evCache.length; i++ ) {
+		if ( evCache[ i ].pointerId === e.pointerId ) {
+			evCache.splice( i, 1 )
+			break
+		}
+	}
+}
+
+const clearZoom = ( image ) => {
+	if ( image ) {
+		image.style.transform = `scale(${scaleMin})`
+		zoomedIn = false
+	}
+}
+
+const toggleZoom = ( e ) => {
+	const image = grabImageFromEvent( e )
+	temp.clientX = null
+	temp.clientY = null
+	temp.translateX = 0
+	temp.translateY = 0
+
+	if ( isImgZoomedIn() ) {
+		image.style.transform = `scale(${scaleMin})`
+		zoomedIn = false
+	} else {
+		image.style.transform = `scale(${scaleMax})`
+		zoomedIn = true
+	}
+}
+
+const zoomStart = ( e ) => {
+	const image = grabImageFromEvent( e )
+	const imageStyle = window.getComputedStyle( image )
```

Comment:
When touching the cue, I got this error.
<img src="https://user-images.githubusercontent.com/2560096/121715101-cfa24200-cade-11eb-9d13-81f371342424.png" width="500"/>